cpp/include/cuml/tree/flatnode.h

```diff
@@ -16,42 +16,71 @@
 
 #pragma once
 
+// We want to define some functions as usable on device
+// But need to guard against thiis file being compiled by a host compiler
+#ifdef __CUDACC__
```

Comment:
Is this header in any non-CUDA code? Quick search in repo revealed that it is always included in CUDA files. In which case, we don't need this guard.